examples/nextjs-with-typescript/pages/MuxPlayer.tsx

```diff
@@ -235,4 +241,41 @@ function MuxPlayerPage() {
   );
 }
 
+function getAvailableControlsCssVars(node) {
+  const cssVars: Record<string, any> = {
+    '--controls': 'none',
+    '--top-controls': 'none',
+    '--center-controls': 'none',
+    '--bottom-controls': 'none',
+  };
+  for (const element of node.querySelectorAll('*')) {
+    if (!element.localName.includes('-')) continue;
+    const matches = element.localName.match(/^([^-]+)-(.*)-(button|range|display)$/);
+    if (!matches) continue;
+    const section = element.closest('[slot="top-chrome"]') ? 'top'
+      : element.closest('[slot="centered-chrome"]') ? 'center' : 'bottom';
+    cssVars[`--${matches[2]}-${matches[3]}`] = 'none';
+    cssVars[`--${section}-${matches[2]}-${matches[3]}`] = 'none';
+  }
+  return Object.keys(cssVars);
+}
+
+const DEFAULT_OPTIONS = {
+  config: { attributes: true, childList: true, subtree: true },
+};
+function useMutationObserver(targetEl, cb, options = DEFAULT_OPTIONS) {
```

Comment:
issue(blocker): Creating a mutation observer hook to manage this introduces too much complexity for an example and shouldn't be necessary. If it is, I suspect we need to revisit why and resolve that.